Notebooks/Pluto/p5.jl

```diff
@@ -0,0 +1,359 @@
+let
+	CSWAP = chain(3, control(1, 2:3=>SWAP)) #The CSWAP gate
+	plot(CSWAP)
+end
+
+# ╔═╡ 96d92a88-4459-11eb-29ee-336cbba060a7
+md"### The Measure Gate"
+
+# ╔═╡ b23f15ca-4459-11eb-3089-f7ea50aaf5de
+md"We already know how to measure the qubits. We can do it in the circuit itself too."
+
+# ╔═╡ c9233378-4459-11eb-0aa9-af1cd8cef6ca
+begin
+	MeasureGate = chain(2, repeat(H, 1:2), Measure(2, locs=1:2))
+	plot(MeasureGate)
+end
+
+# ╔═╡ 1615f882-445a-11eb-3a16-c76bb2305ff5
+md"Note that now, when we measure them using the measure block, the output remains unchanged, even though we should've a 25% chance of getting $ |00〉, |01〉, |10〉 $ or $ |11〉 $."
+
+# ╔═╡ 9626495a-445a-11eb-0a2d-2168c66ad1e0
+zero_state(2) |> MeasureGate |> r->measure(r, nshots=1024)
+
+# ╔═╡ cd54fd7c-445a-11eb-0015-bd814cd6293f
+md"Without the Measure gate,"
+
+# ╔═╡ dbe0b9ee-445a-11eb-3526-29900707092e
+zero_state(2) |> repeat(2, H, 1:2) |> r->measure(r, nshots=1024)
+
+# ╔═╡ 06d1ed92-4485-11eb-3485-d1be36b9f68b
+md"### The LabeledBlock"
+
+# ╔═╡ 15f64d2c-4485-11eb-3749-a33d80bf5b6b
+md"Its used for easily plotting circuits as boxes for simpler visualization."
+
+# ╔═╡ 2c8d9af4-4485-11eb-238e-b36d910466fc
+let
+	a = chain(3, repeat(H, 1:2), put(3=>X))
+	b = chain(2, repeat(Y, 1:2))
+	circuit = chain(3, put(1:3 => label(a, "circuit a")), put(2:3 => label(b, "circuit b")))
+	plot(circuit)
+end
+
+# ╔═╡ f5e2eb8a-4484-11eb-0957-41f14bbfc3d9
+md"### Daggered Block"
+
+# ╔═╡ 00b09d5a-4485-11eb-2857-011786a85254
+md"We use Daggered block to build circuits which undo the effects of a particular circuit.
+
+Let's take an example. Remember Bell Circuit?"
+
+# ╔═╡ ec2e6fb4-4485-11eb-351d-4781925a47d3
+begin
+	bellcircuit = chain(2, put(1=>H), control(1, 2=>X))
+	plot(bellcircuit)
+end
+
+# ╔═╡ 1791bfd0-4486-11eb-0107-e345d88b2ddb
+state(zero_state(2) |> bellcircuit)
+
+# ╔═╡ 61a0a0a0-4486-11eb-2f6a-750de59d5988
+md"Building the Reverse Bell Circuit is as easy as,"
+
+# ╔═╡ 771faeba-4486-11eb-07ac-4dc4d52d4726
+state(zero_state(2) |> bellcircuit |> Daggered(bellcircuit))
+
+# ╔═╡ b7d76bd4-4486-11eb-3565-ab0ef9238acf
+md"Plotting the Daggered Block is a bit tricky! YaoPlots doesn't support DaggeredBlock yet. There are two alternatives to this."
+
+# ╔═╡ 39890364-4488-11eb-22d7-33a8db179379
+plot(Daggered(bellcircuit))
+
+# ╔═╡ 4eb6a852-4488-11eb-0e42-7be38fca7571
+md"One way is to use the label block"
+
+# ╔═╡ 11a084ac-4487-11eb-30ea-539b38e0231f
+let
+	reversebellcircuit = put(2, 1:2 => label(Daggered(bellcircuit), "Bell\n Circuit†"))
+	plot(reversebellcircuit)
+end
+
+# ╔═╡ 49ed44ca-4488-11eb-07dd-27f8ca6a1d41
+md"Another way, is to use ` ' `."
+
+# ╔═╡ 6325cf34-4488-11eb-0415-1d1f2fa0f1ba
+plot(bellcircuit')
+
+# ╔═╡ 6f5420f8-4488-11eb-3cf9-03675dc3c8db
+md"Whats the difference between the DaggeredBlock and adjoint ` ' ` ?
+
+One is a function of Yao, while another of Base julia. Both perform the same operation on their input, but DaggeredBlock is many times faster."
```

Comment:
```suggestion
One is a function of Yao, while another of julia `Base`. Both perform the same operation on their input"
```

note `Base` is a julia module, should be in code block, as well as other things. It's not true that `DaggerBlock` is many times faster, actually in some cases where the gates are hermitian `adjoint` is a little bit faster, this is because `adjoint` will auto simplify the adjoint expression, and only put `DaggeredBlock` to what needs it.

I don't suggest anyone use `DaggeredBlock`, it should be something more internal (I can't remember why we export it but maybe we shouldn't). One should always use the adjoint syntax sugar.

Reply:
Maybe you need to remove the introduction of `DaggeredBlock` or at least emphasize the usage of `'`.

Reply:
I remember making a circuit to test the speeds... 
Well, if its not faster, I think I'll replace the Daggered block with '